pylint/checkers/imports.py

```diff
@@ -697,9 +700,10 @@ def _add_imported_module(self, node, importedmodname):
         # itself using this condition (the level will be bigger or
         # if the same module is named as the package, it will be different
         # anyway).
-        if isinstance(node, astroid.ImportFrom):
-            if node.level and node.level > 0 and base == '__init__':
-                return
+        # XXX why is this done?  It hides packages from the imprort graph
```

Comment:
This one was added by https://github.com/PyCQA/pylint/commit/bcf801f5e7fadb449c363c7acc14390286dceb7c, but the check should only apply to `import-self` warning that gets emitted a bit down this function, not for the import graph update. Ideally we should move the `add_message` call and this filter into their own function, and call that instead when we want to emit the error instead of calling `_add_imported_module`.

Reply:
This looks like it could simply be removed.  The unit test pass with the code commented out, and printing the the result of the `get_module_part` call reveals it no longer returns `dummy` for `dummy.dummy.Dummy`, instead it returns `dummy.dummy`.